src/sbl_filing_api/services/submission_processor.py

```diff
@@ -116,14 +116,44 @@ async def validate_and_update_submission(
 def build_validation_results(results: ValidationResults):
     val_json = df_to_dicts(results.findings, settings.max_json_records, settings.max_json_group_size)
     if results.phase == ValidationPhase.SYNTACTICAL:
-        val_res = {"syntax_errors": {"count": results.single_field_count, "details": val_json}}
+        val_res = {
+            "syntax_errors": {
+                "single_field_count": int(results.error_counts.single_field_count),
+                "multi_field_count": int(
+                    results.error_counts.multi_field_count
+                ),  # this will always be zero for syntax errors
+                "register_count": int(
+                    results.error_counts.register_count
+                ),  # this will always be zero for syntax errors
+                "total_count": int(results.error_counts.total_count),
+                "details": val_json,
+            }
+        }
     else:
         errors_list = [e for e in val_json if e["validation"]["severity"] == Severity.ERROR]
         warnings_list = [w for w in val_json if w["validation"]["severity"] == Severity.WARNING]
         val_res = {
-            "syntax_errors": {"count": 0, "details": []},
-            "logic_errors": {"count": sum([len(v["records"]) for v in errors_list]), "details": errors_list},
-            "logic_warnings": {"count": sum([len(v["records"]) for v in warnings_list]), "details": warnings_list},
+            "syntax_errors": {
+                "single_field_count": 0,
+                "multi_field_count": 0,
+                "register_count": 0,
+                "total_count": 0,
+                "details": [],
+            },
+            "logic_errors": {
+                "single_field_count": int(results.error_counts.single_field_count),
```

Comment:
`results.error_counts` is a `Counts` object, right? And the attributes in the objects are ints? Why do we need to wrap with `int()`?

Reply:
@lchen-2101 They're being returned as `int64` on the validator side and the JSON is not serializable as a result. So they needed to be cast as `int` until we can remedy it validator-side.

Reply:
The data-validator is doing a sum([int64]) which is returning an int64 value of the total.  This type isn't serializable to it's causing an issue in the filing-api.  I wrote a story to fix the data-validator side.  We can either fix this after that is done and merged in (leaving the int() cast here) or knock out the data-validator story and remove the casts.

Reply:
yeap, just saw your comment on the PR :joy:

Reply:
do we want to hold off on this one then?